internal/rbd/controllerserver.go

```diff
@@ -591,9 +591,22 @@ func (cs *ControllerServer) createVolumeFromSnapshot(
 
 		return err
 	}
+	deleteClone := true
+
```

Comment:
May be move the defer here and set the `deleteClone = false` once all the operations succeeds

```
    deleteClone := true                                                              
    defer func() {                                                                   
        if deleteClone {                                                             
            err = librbd.RemoveImage(rbdVol.ioctx, rbdVol.RbdImageName)                      
            if err != nil {                                                          
                log.ErrorLog(ctx, "failed to delete clone image %q: %v", rbdVol, err)
            }                                                                        
        }                                                                            
    }()                                                                              
    err = rbdVol.unsetAllMetadata(k8s.GetSnapshotMetadataKeys())
    [...]
    err = parentVol.copyEncryptionConfig(&rbdVol.rbdImage, true)
    [...]
    err = rbdVol.expand()
    [...]
    deleteClone = false

    return nil
}
```

Reply:
you dont need a variable. do `if err!=nil` check in defer and delete the clone on failure.